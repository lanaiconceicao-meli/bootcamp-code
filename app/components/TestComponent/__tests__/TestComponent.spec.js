const React = require('react');
const { render } = require('@testing-library/react');
const TestComponent = require('../TestComponent');

describe('snapshot test', () => {
  const i18n = { gettext: text => text };
  test('TestComponent snapshot', () => {
    const { asFragment } = render(<TestComponent isArray={[]} i18n={i18n} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('TestComponent with array snapshot', () => {
    const { asFragment } = render(<TestComponent isArray={['agua', 'terra', 'ar', 'fogo']} i18n={i18n} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
