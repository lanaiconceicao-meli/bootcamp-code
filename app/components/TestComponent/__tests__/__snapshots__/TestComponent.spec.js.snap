// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test TestComponent snapshot 1`] = `
<DocumentFragment>
  <h1>
    n√£o ex
  </h1>
</DocumentFragment>
`;

exports[`snapshot test TestComponent with array snapshot 1`] = `
<DocumentFragment>
  <h4>
    agua
  </h4>
  <h4>
    terra
  </h4>
  <h4>
    ar
  </h4>
  <h4>
    fogo
  </h4>
</DocumentFragment>
`;
